lib/travis/logs/aggregate.rb

```diff
@@ -24,10 +27,18 @@ def run
       end
 
       def aggregate_logs
-        Travis::Logs::Services::AggregateLogs.run
+        exclusive do
+          Travis::Logs::Services::AggregateLogs.run
+        end
       rescue Exception => e
```

Comment:
Seems like this `rescue` should probably be in the Exclusive as a begin block since we're really wanting to capture exceptions from the run, not the lock. If the lock exceptions we really want to know about that.